src/pools/abstract-pool.ts

```diff
@@ -26,34 +43,72 @@ export interface PoolOptions<Worker> {
    */
   exitHandler?: ExitHandler<Worker>
   /**
-   * This is just to avoid not useful warnings message, is used to set `maxListeners` on event emitters (workers are event emitters).
+   * This is just to avoid non-useful warning messages.
+   *
+   * Will be used to set `maxListeners` on event emitters (workers are event emitters).
    *
    * @default 1000
+   * @see [Node events emitter.setMaxListeners(n)](https://nodejs.org/api/events.html#events_emitter_setmaxlisteners_n)
    */
   maxTasks?: number
 }
 
+/**
+ * Internal poolifier pool emitter.
+ */
 class PoolEmitter extends EventEmitter {}
 
+/**
+ * Basic pool abstraction used for common logic between poolifier pools.
+ *
+ * @template Worker Type of worker which manages this pool.
+ * @template Data Type of data sent to the worker.
+ * @template Response Type of response of execution.
+ */
 export abstract class AbstractPool<
   Worker extends IWorker,
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   Data = any,
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   Response = any
 > implements IPool<Data, Response> {
+  /**
+   * List of current workers.
+   */
   public readonly workers: Worker[] = []
+
+  /**
+   * ID for the next worker.
+   */
   public nextWorker: number = 0
```

Comment:
Should we rename this property?

Reply:
You can but keep a `get id() { return this messageId }`

Reply:
@pioardi Yeah we could do that for backward compatibility reasons
But it as we already changed so much for the next major release, I think it's not necessary

Reply:
#134